soc/arm/ti_k3/am62x_m4/soc.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Texas Instruments Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <soc.h>
+
+#define ADDR_TRANSLATE_RAT_BASE_ADDR		(0x044200000u)
+#define ADDR_TRANSLATE_REGIONS			(4u)
```

Comment:
nitpick: This doesn't really need to be defined.
If you remove the size from the array definition, the initialiser list defines the size:
`struct address_trans_region_config region_config[] =  {`
Then you can use `ARRAY_SIZE(region_config)` in the call to `sys_mm_drv_ti_rat_init()`.
This will help prevent `ADDR_TRANSLATE_REGIONS` from being a different size to the initialiser list